test/Sema/throwing_super_init.swift

```diff
@@ -0,0 +1,24 @@
+// RUN: %target-typecheck-verify-swift
+
+class ThrowingInitSuperclass {
+  init() throws { }
+}
+
+// Implicitly calling the super.init IS possible here because the initializer
+// is a throwing initializer.
+class ThrowingInitClass: ThrowingInitSuperclass {
+  init(simpleArgument: Int) throws { }
+}
+
+// Implicitly calling the super.init IS NOT possible here because the
+// initializer is not a throwing initializer.
+class NonThrowingInitClass: ThrowingInitSuperclass {
+  init(simpleArgument: Int) { } // expected-error {{missing call to superclass's initializer; 'super.init' is a throwing initializer and requires either an explicit call or that this initializer is also marked as 'throws'}}
+}
+
+// Implicitly calling the super.init IS NOT possible here because the
+// initializer is a rethrowing initializer, which means it can only
+// rethrow errors from its parameters.
+class RethrowingInitClass: ThrowingInitSuperclass {
+  init(throwingArgument: () throws -> Void) rethrows { } // expected-error {{missing call to superclass's initializer; 'super.init' is a throwing initializer and requires either an explicit call or that this initializer is also marked as 'throws'}}
```

Comment:
Can we actually suggest an explicit call in this case or only viable option is to change `rethrows` to `throws`?

Reply:
Is the only way to make it work in both cases:

```
do {
  try super.init()
} catch {
   fatalError(...)
}
```

Or something similar?

Reply:
> Can we actually suggest an explicit call in this case or only viable option is to change `rethrows` to `throws`?

By "suggest" do you mean adding a fixit? Because the diagnostic message already suggests an explicit call, or am I missing something?

> Is the only way to make it work in both cases:
> 
> ```
> do {
>   try super.init()
> } catch {
>    fatalError(...)
> }
> ```
> 
> Or something similar?

I believe this is a possible solution, yeah, but I can think of two alternatives: disabling the error propagation via `try!` and making use of a failable initializer, enabling a `return nil` on the catch block.

Reply:
I was asking whether saying that "explicit" call is viable here from the developer perspective, I guess with `try!` failable initializer and `do { } catch { ... }` there are options.

Reply:
> Can we actually suggest an explicit call in this case or only viable option is to change `rethrows` to `throws`?

The intended takeaway from this error message is that in order to proceed in the given direction you either have to rethink the `rethrows` or explicitly call `super.init()` *in a way the compiler would be happy about*.